pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java

```diff
@@ -270,6 +275,7 @@ public String load(String instanceId) {
       _lineageUpdaterLocks[i] = new Object();
     }
     _lineageManager = lineageManager;
+    _executorService = executorService;
```

Comment:
looks like no need for _executorService and _connMgr, as you provide the `_consumingSegmentInfoReader` to pinotHelixResMgr directly